### ISSUE TITLE
! Donated amount gets changed when submitting to Mollie Payment Gateway

### ISSUE THREAD

**Issue by Marcoevich** [impress-org/givewp#4481]
# Bug Report

## User Story

As a donor I want to donate €680. Now when submitting the form, everything says I am paying €680. But when reaching Mollie Payment Gateway, the amount is changed to €68,-, so the last zero is missing. This can potentially cost a lot of money. If I try to pay €6800, then the last 2 zeroes are missing in the Mollie Payement Gateway.

## Current Behavior
Ending zeroes are missing / detached when communicating the donation amount to Mollie Payment Gateway

## Expected Behavior
The exact amount that I fill in the form should be communicated to the payment gateway

## Bug Type
- [x] This bug describes functionality that once worked as expected in version X.X.X. (don't know the previous version)
- [ ] This bug describes functionality that never worked as expected.
- [ ] I am not sure whether this functionality ever worked as expected.

## Steps to Reproduce
1. Donate here: https://louterbloemen.nl/bouw-mee/
2. Enter a custom amount with ending zeroes
3. See the amount gets changed when you enter Mollie Payment Gateway

## Visuals
This video shows what happens: https://www.screencast.com/t/pBq4Vq6Uj1t4 

## Acceptance Criteria
- [x] e.g. Something happens when an action is taken.
- [ ] e.g. Something does not happen when an action is taken.
- [ ] e.g. Fixing behavior in Component A does not affect existing behavior in Component B.

## Environment

<details>
    <summary>Operating System</summary>
    <ul>
        <li>Platform: Microsoft Windows</li>
        <li>Version: Windows 10, version 1909 64-bit</li>
    </ul>
</details>

<details>
    <summary>Browser</summary>
    <ul>
        <li>Name: Chrome</li>
        <li>Version: 79.0.3945.130</li>
    </ul>
</details>

<details>
    <summary>WordPress System Info</summary>
    ### WordPress Environment ###

Home URL: https://louterbloemen.nl
Site URL: https://louterbloemen.nl
WP Version: 5.3.2
WP Multisite: –
WP Memory Limit: 768 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: nl_NL
Permalink Structure: /%postname%/
Show on Front: page
Page on Front: Home (#62)
Page for Posts: Unset
Table Prefix Length: wprc_
Table Prefix Length: 5
Table Prefix Status: Acceptable
Admin AJAX: Accessible
Registered Post Statuses: publish, future, draft, pending, private, trash, auto-draft, inherit, request-pending, request-confirmed, request-failed, request-completed, acf-disabled, refunded, failed, revoked, cancelled, abandoned, processing, preapproval, tribe-ea-success, tribe-ea-failed, tribe-ea-schedule, tribe-ea-pending, tribe-ea-draft, tribe-ignored, give_subscription

### Server Environment ###

Hosting Provider: DBH: localhost, SRV: louterbloemen.nl
TLS Connection: Connection uses TLS 1.2
TLS Connection: Probably Okay
Server Info: Apache
PHP Version: 7.3.14
PHP Post Max Size: 128 MB
PHP Time Limit: 500
PHP Max Input Vars: 3000
PHP Max Upload Size: 128 MB
cURL Version: 7.59.0, OpenSSL/1.0.2r
SUHOSIN Installed: –
MySQL Version: 5.6.40
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
gzip: ✔
GD Graphics Library: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### GiveWP Configuration ###

GiveWP Version: 2.5.13
GiveWP Cache: Enabled
Database Updates: All DB Updates Completed.
Database Tables: ✔ wprc_give_donors -  1.0✔ wprc_give_donormeta -  1.0✔ wprc_give_comments -  1.0✔ wprc_give_commentmeta -  1.0✔ wprc_give_sessions -  1.0✔ wprc_give_logs -  1.0✔ wprc_give_logmeta -  1.0✔ wprc_give_formmeta -  1.0✔ wprc_give_sequential_ordering -  1.0✔ wprc_give_donationmeta -  1.0
GiveWP Cache: Enabled
GiveWP Cache: ✔New Donation✔Donation Receipt❌New Offline Donation❌Offline Donation Instructions✔New User Registration✔User Registration Information✔Donation Note✔Email access❌Daily Email Report✔Weekly Email Report✔Monthly Email Report❌Renewal Receipt Email❌Renewal Receipt Email❌Subscription Cancelled Email❌Subscription Cancelled Email❌Subscription Completed Email❌Subscription Reminder Email✔Subscriptions Email Access✔Subscription Payment Failed Email
Upgraded From: 2.5.3
Test Mode: Disabled
Currency Code: EUR
Currency Position: Before
Decimal Separator: ,
Thousands Separator: .
Success Page: https://louterbloemen.nl/donatie-bevestiging/
Failure Page: https://louterbloemen.nl/donatie-mislukt/
Donation History Page: https://louterbloemen.nl/donatiegeschiedenis/
GiveWP Forms Slug: /donations/
Enabled Payment Gateways: Mollie Gateway
Default Payment Gateway: Mollie Gateway
PayPal IPN Verification: Enabled
PayPal IPN Notifications: N/A
Donor Email Access: Enabled
Stripe Webhook Notifications: N/A

### Active GiveWP Add-ons ###

Give - Email Reports: ✔ Licensed – by GiveWP – 1.1.4
Give - Fee Recovery: ✔ Licensed – by GiveWP – 1.7.7
Give - Google Analytics Donation Tracking: ✔ Licensed – by GiveWP – 1.2.4
Give - MailChimp: ✔ Licensed – by GiveWP – 1.4.5
Give - Manual Donations: ✔ Licensed – by GiveWP – 1.4.6
Give - Mollie Gateway: ✔ Licensed – by GiveWP – 1.2.3
Give - PDF Receipts: ✔ Licensed – by GiveWP – 2.3.6
Give - Recurring Donations: ✔ Licensed – by GiveWP – 1.9.10

### Other Active Plugins ###

404page - your smart custom 404 error page: by Peter Raschendorfer – 11.2.1
ACF Content Analysis for Yoast SEO: by Thomas Kräftner, ViktorFroberg, marol87, pekz0r, angrycreative, Team Yoast – 2.3.0
Activity Log: by Activity Log Team – 2.5.2
Advanced Custom Fields PRO: by Elliot Condon – 5.8.7
Analytics Spam Blocker: by Arnan de Gans – 2.9.2
Classic Editor: by WordPress Contributors – 1.5
Coming Soon Page, Under Construction & Maintenance Mode by SeedProd: by SeedProd – 5.1.0
Cookie Notice: by dFactory – 1.2.50
Download Monitor: by Never5 – 4.4.2
Duplicate Page: by mndpsingh287 – 4.0
Fusion Builder: by ThemeFusion – 2.1.2
Fusion Core: by ThemeFusion – 4.1.2
Fusion White Label Branding: by ThemeFusion – 1.1.3
Google Analytics Dashboard for WP (GADWP): by ExactMetrics – 5.3.10
Gravity Forms: by rocketgenius – 2.4.17
Loco Translate: by Tim Whitlock – 2.3.1
MC4WP: Mailchimp for WordPress: by ibericode – 4.7.4
Members: by Justin Tadlock – 2.2.0
Ninja Forms: by The WP Ninjas – 3.4.22
Ninja Forms - Conditional Logic: by The WP Ninjas – 3.0.26
Ninja Forms - Layout & Styles: by The WP Ninjas – 3.0.28
PWA: by PWA Plugin Contributors – 0.3.0
SG Optimizer: by SiteGround – 5.4.3
The Events Calendar: by Modern Tribe, Inc. – 4.9.5
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com, DavidAnderson – 1.16.21
Wordfence Security: by Wordfence – 7.4.5
WP Extra File Types: by Davide Airaghi – 0.4.4.1
WP Flickity: by PaoloFalomo – 0.5.1
WP Mail SMTP: by WPForms – 1.8.1
WP Sitemap Page: by Tony Archambeau – 1.6.2
Yoast SEO: by Team Yoast – 12.9.1

### Inactive Plugins ###

Category Tag Pages: by Marzio Carro – 1.0
Duplicator: by Snap Creek – 1.3.24
Enable Media Replace: by ShortPixel – 3.3.7
Give - Form Field Manager: by GiveWP – 1.4.4
Give - Tributes: by GiveWP – 1.5.6
Link Checker: by Marco Beierer – 1.18.0
Ninja Forms - File Uploads: by The WP Ninjas – 3.0.22
Ninja Forms - Multi-Part Forms: by The WP Ninjas – 3.0.26
Regenerate Thumbnails: by Alex Mills (Viper007Bond) – 3.1.2
WP-Optimize - Clean, Compress, Cache: by David Anderson, Ruhani Rabin, Team Updraft – 3.0.16

### Theme ###

Name: Louterbloemen
Version: 1.0.0
Author URL: https://www.planatstudio.com
Child Theme: Yes
Parent Theme Name: Avada
Parent Theme Version: 6.1.2
Parent Theme Author URL: http://themeforest.net/user/ThemeFusion
</details>

**Comment by ravinderk** [comment]
@Marcoevich  I am unable to reproduce this issue. Please check this video: https://www.loom.com/share/21fc534bf8b94af58a4a5423ce3a67d5

**Comment by DevinWalker** [comment]
I just checked as well - did something change on your site where you fixed the issue?

**Comment by Marcoevich** [comment]
Hmm I checked the site activity log. Nothing changed I see only a whole lot of donation tests from different locations. I've checked from home and even RDP'd back into my work computer to check from there. Issue seems to be resolved on it's own. Could it be a temporary issue at the payment gateway?

I find it scary to have these sort of issues and not knowing what caused it. I hope it was temporary and doesn't happen again.
